File: app/src/main/java/com/sapuglha/tictactoe/di/ActivityBuilder.java
package com.sapuglha.tictactoe.di;

import com.sapuglha.tictactoe.ui.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector
    abstract MainActivity bindMainActivity();
}

File: app/src/main/java/com/sapuglha/tictactoe/di/ActivityBuilder.kt
package com.sapuglha.tictactoe.di

import com.sapuglha.tictactoe.ui.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector
    internal abstract fun bindMainActivity(): MainActivity
}
